docs/authentication.md

```diff
@@ -0,0 +1,76 @@
+# Authentication
+
+This page documents the different ways that authentication can be configured for
+`secrets-store-csi-driver-provider-gcp`.
+
+## Workload Identity (default)
+
+The identity of the pod the secrets are mounted onto.
+
+When the GCP provider receives a `Mount` request it obtains a Kubernetes
+Service Account token for the associated pod. This token is then exchanged for
+a Google authentication token using the Kubernetes Service Account [Workload
+Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity)
+annotations.
+
+## `self-adc`
```

Comment:
Should we just call this `adc`?

Reply:
I didn't want it to be just `adc` because i thought that might get confusing with workload identity since that is also used in the context of application default creds.

Do you think `provider-adc` might be better?

Reply:
I like `provider-adc`, or maybe even `secret-provider-adc` if it's not too long.